src/main/java/mate/academy/controller/AuthController.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.UserRegistrationRequestDto;
+import mate.academy.dto.UserResponseDto;
+import mate.academy.exception.RegistrationException;
+import mate.academy.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
+            throws RegistrationException {
+        return userService.register(request);
+    }
```

Comment:
Consider to handle the RegistrationException using the GlobalExceptionHandler or @ExceptionHandler